src/auth/auth.service.ts

```diff
@@ -28,17 +28,31 @@ export class AuthService {
 
   // Sign user in by email and password
   async login(credentials: LoginUserDto): Promise<AccessToken> {
-    const user = await this.usersService.findOneUnsanitized({email: credentials.email});
-    const passwordHash = user.password;
-    const isCorrectPassword = await bcrypt.compare(credentials.password, passwordHash);
-    if(!isCorrectPassword || !passwordHash) {
+    const user = await this.validateUser(credentials);
+    if(!user) {
       throw new UnauthorizedException();
     }
 
     // Add a username in the JWT token
-    const payload = {username: user.username, isAdmin: user.isAdmin};
+    const payload = {
+      sub: user['id'],
```

Comment:
User ID should not be disclosed. Should we use username, or change the schema to use UUIDs even for users?

Reply:
It's not exactly disclosing since we are putting this payload into a JWT token, no? Anyways, I do prefer the user id to be a uuid, it seems like a clean way to do things.